homeassistant/components/squeezebox/browse_media.py

```diff
@@ -57,63 +61,113 @@
     "Tracks": MediaType.TRACK,
     "Playlists": MediaType.PLAYLIST,
     "Genres": MediaType.GENRE,
+    "Favorites": MediaType.TRACK,
+}
+
+LIBRARY_ICONS = {
+    "Artists": "html/images/artists.png",
+    "Albums": "html/images/albums.png",
+    "Tracks": "html/images/musicfolder.png",
+    "Playlists": "html/images/playlists.png",
+    "Genres": "html/images/genres.png",
+    "Favorites": "html/images/favorites.png",
 }
 
 BROWSE_LIMIT = 1000
 
 
+def _lms_prefix(player):
+    return (player.generate_image_url_from_track_id("")).split("music/")[0]
+
+
 async def build_item_response(entity, player, payload):
     """Create response payload for search described by payload."""
+
     internal_request = is_internal_request(entity.hass)
 
     search_id = payload["search_id"]
     search_type = payload["search_type"]
 
     media_class = CONTENT_TYPE_MEDIA_CLASS[search_type]
 
-    if search_id and search_id != search_type:
-        browse_id = (SQUEEZEBOX_ID_BY_TYPE[search_type], search_id)
-    else:
-        browse_id = None
+    children = None
 
-    result = await player.async_browse(
-        MEDIA_TYPE_TO_SQUEEZEBOX[search_type],
-        limit=BROWSE_LIMIT,
-        browse_id=browse_id,
-    )
+    if search_type == "Favorites":
+        _command = ["favorites"]
+        _command.extend(["items", "0", "100"])
```

Comment:
result = await self.async_query("result = await self.async_query("favorites", "items")
result["count"] rather than limit to 100?

Reply:
Hi @pssc We don't know the count at that point - until after we run the command to get the favorites.  I just picked an arbitrary number larger than I thought people would have.  To get the count first, I could do a "favorites items" which just returns the count, but that's just an extra call I think.  I could change the number to 1000 (i.e. basically just all of them).  Thoughts@

Reply:
Yes if we're going to set it to 1000, we might as well use browse_limit, but that's more to limit the number of tracks etc. where the number could be very large.  Certainly no harm in using it here as well though.  Think it's likely a sensible change - or at least there's no harm doing it IMHO.